pkg/collector/corechecks/servicediscovery/events.go

```diff
@@ -80,7 +79,6 @@ func (ts *telemetrySender) newEvent(t eventType, svc serviceInfo) *event {
 func newTelemetrySender(sender sender.Sender) *telemetrySender {
 	return &telemetrySender{
 		sender:   sender,
-		time:     realTime{},
```

Comment:
Because of this change, the linter on windows and mac complains the `timer` interface from `servicediscovery.go` is unused (because its only used on the build constrained `impl_linux.go` file).

We need to either move the interface to the `impl_linux.go` file (and re-generate the mock), or write some unit test in the non build-constrained files that uses this interface.

Reply:
argh, ok, I will move the interface to make the linter happy 🤷